common/core/core_error.c

```diff
@@ -0,0 +1,150 @@
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+
+#include "core_error.h"
+
+#include "constant_texts.h"
+#include "events.h"
+#include "p0_events.h"
+#include "ui_message.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+/**
+ * @brief Display error message to user set by core operations.
+ *
+ * @details When a core operation faces a fatal error, it can set an error
+ * message using using @ref mark_core_error_screen before exiting the flow. This
+ * API displays that error message. NOTE: P0 events are ignored in this API, as
+ * this could also be used to display P0 errors
+ */
+static void display_core_error();
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+static char core_error_msg[60] = {0};
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+static void display_core_error() {
+  if (0 == strnlen(core_error_msg, sizeof(core_error_msg)))
+    return;
+
+  evt_status_t status = {0};
+  message_scr_init(core_error_msg);
+
+  do {
+    status = get_events(EVENT_CONFIG_UI, INIFINITE_WAIT_TIMEOUT);
```

Comment:
Hey Aman, this could be risky if USB host sends an abort while we are stuck in this loop.. so it's better if we update device idle state so that USB aborts are blocked before entering this loop